packages/dom/src/utils/assert-is-defined.ts

```diff
@@ -2,7 +2,10 @@ export function assertIsDefined< T >(
 	val: T,
 	name: string
 ): asserts val is NonNullable< T > {
-	if ( val === undefined || val === null ) {
+	if (
+		process.env.NODE_ENV === 'development' &&
```

Comment:
We need 
```suggestion
		process.env.NODE_ENV !== 'production' &&
```
to make unit tests pass.